### PR TITLE
gh-95: Add photometric RSD window for PositionsTracer

### LINKED ISSUE #95: Include photometric RSD
### Description
Include RSD in the angular photometric galaxy clustering

### Goals/Acceptance Criteria
- [x] **Goal 1**: include this systematic effect to keep functionality with respect old CLOE
- [x] **Goal 2**: Should not make the code slower

### Steps to Reproduce (if applicable)
TBA

### Expected Outcome
Make RSD available

### Current Behavior (if applicable)
Right now, this effect is not present

### Additional Notes
TBA

### Assignees
- TBA

### PR DESCRIPTION
**Pull request by ktanidis**
### ✅ Summary

Add optional photometric linear RSD contribution for `PositionsTracer` using shifted redshift window based on (Tanidis & Camera 2019). Enabled via `include_rsd=True` so it affects `POS×POS` and `POS×SHE` consistently.

### 🔄 Changes

- Additions in `photo.py`. Construct the linear RSD window function by adding the helping functions: `_L_coeffs`, `_alpha_coeffs`, `_interp_linear_2d_queries` and `get_photo_rsd`. In the initialization of the `PositionsTracer` class there is now the new `include_rsd` option. The new function `get_window_rsd` with API is introduced in this class.
- Additions in  `angular_two_point.py`. These helping functions are introduced: 1) The `_growth_rate_on_grid`  and 2) the functions `Cl_int_liz_jz`, `Cl_int_iz_ljz`, `Cl_int_liz_ljz` when there is multipole dependence if the RSD window function is present. Instantiate the `need_rsd` in the `get_Cl` function and built the angular power spectra based on the helping and the speed-up functions. 
- It doesn't slow down the code
### 🛠 How to Test

Introducing  the new argument `include_rsd=True` in the `PositionsTracer` activates the linear RSD in the photometric galaxy clustering. 

### 📝 Documentation

- [ ] This PR updates documentation
- [ ] This PR does not require documentation changes
- [ ] Issue created for documentation update: Don't forget to link the task!

### 🏗 Related Issues

Resolves #95 

### 📸 Screenshots (if applicable)

Measured on CPU after JAX warmup (100 ell values, 6 tomographic bins):
- POS×POS get_Cl: no RSD 0.0327s, w/ RSD 0.0359s (~1.10×)
- SHE×POS get_Cl: no RSD 0.0504s, w/ RSD 0.0457s (~0.91×, within run-to-run scatter)

Ratio of spectra w/o RSD to spectra w/RSD for POS×POS and SHE×POS:

image

image

The percentage of power gained at large scales ($\ell\lt$ 200) due to RSD is consistent with findings in the literature for Euclid-like tomographic bins (Fig. 1 of https://arxiv.org/pdf/2309.00052).

### 📌 Additional Notes

---

### ✅ PR Checklist for Developers

- [x] I have titled this PR before merging as "gh-#:", where "#" represents the task it closes
- [x] I have run locally pre-commit using `pre-commit run --all-files`
- [x] I have tested my changes locally
- [x] No new warnings or errors introduced
- [x] I have updated documentation (if applicable)
- [x] My changes do not introduce breaking changes (i.e: the package still gets installed)
- [x] I have added unit tests (if applicable)
- [x] I have consistently updated the GitHub information for the project, including milestones, task types, and other relevant details.

### ✅ PR Checklist for Reviewers

- [ ] The next PR targets the correct branch
- [ ] CI tests have run and passed for the latest commit on the source branch
- [ ] Check that the code can still be installed if new packages are imported
- [ ] If necessary, the notebooks in the `playground` will be updated in a corresponding follow-up PR
- [ ] Coverage percentage is retained or increased
- [ ] Quality of new/changed code is acceptable
- [ ] Quality of new/changed unit tests is acceptable
- [ ] No data files have been included in the commits
- [ ] Implementation follows the agreed task description point by point
- [ ] Check that any added folder/file has been added to the `README.md` file
- [ ] Check that the documentation has been updated accordantly
- [ ] Check that the corresponding branch has been deleted after merging. If not, delete it

### CODE REVIEW

**Comment by gcanasherrera** [comment]
@marcobonici can you check the JAX side on the summary_statistics side?

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -19,6 +19,9 @@ jobs:
         with:
           python-version: "3.10"
 
+      - name: Upgrade packaging tools
```
Comment (gcanasherrera) [inline .github/workflows/ci.yaml]: this is not necessary, please remove.

Comment (ktanidis) [reply .github/workflows/ci.yaml]: I removed it, in the latest version.

**Comment by gcanasherrera** [comment]
@ktanidis sorry for the delay on this. Do you have any plots to show the computation? Thanks

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -42,7 +42,7 @@ jobs:
 
       - name: Install external prereq using pip
         run: |
-          pip install camb jax classy==3.3.1
+          python -m pip install camb jax classy==3.3.1
```
Comment (gcanasherrera) [inline .github/workflows/ci.yaml]: remove this python thing

File: cloelib/observables/photo.py [hunk 3]
```diff
@@ -21,6 +21,96 @@
...
+
+    # zero outside bounds
+    oob = (xq_lz < chi[0]) | (xq_lz > chi[-1])
+    out = np.where(oob[None, :, :], 0.0, out)
+
+    return np.transpose(out, (1, 0, 2))
+
+
+@jax.jit
+def photo_rsd_window(ells, chi, S_bin_z):
```
Comment (gcanasherrera) [inline cloelib/observables/photo.py]: can you change the name to get_window_photo_rsd? For consistency

File: cloelib/observables/photo.py [hunk 4]
```diff
@@ -194,6 +284,7 @@ def __init__(
         z: np.ndarray,
         galaxy_bias_model: str,
         nuisance_params: dict,
+        include_rsd: bool = False,  # NEW
```
Comment (gcanasherrera) [inline cloelib/observables/photo.py]: remove the # NEW (AI leftover)

File: cloelib/observables/photo.py (hunk 3 above)
Comment (gcanasherrera) [reply cloelib/observables/photo.py]: Maybe it's better, actually `get_photo_rsd`

**Comment by ktanidis** [comment]
> @ktanidis sorry for the delay on this. Do you have any plots to show the computation? Thanks

No worries @gcanasherrera. I have now included them in the PR description. Thanks!

**Comment by gcanasherrera** [comment]
Thanks @ktanidis ! Plots look neat! 

Just few name changes requests.

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (ktanidis) [reply .github/workflows/ci.yaml]: Done!

File: cloelib/observables/photo.py (hunk 3 above)
Comment (ktanidis) [reply cloelib/observables/photo.py]: Done!

File: cloelib/observables/photo.py (hunk 4 above)
Comment (ktanidis) [reply cloelib/observables/photo.py]: Done!
